Mage.Sets/src/mage/cards/p/PlarggDeanOfChaos.java

```diff
@@ -0,0 +1,198 @@
+            Card eligibleCard = null;
+            for (Card card : controller.getLibrary().getCards(game)) {
+                toReveal.add(card);
+                if (!card.isLand() && !card.isLegendary() && card.getConvertedManaCost() < 4) {
+                    eligibleCard = card;
+                    break;
+                }
+            }
+            controller.revealCards(source, toReveal, game);
+            if (eligibleCard != null
+                    && controller.chooseUse(Outcome.PlayForFree, "Cast " + eligibleCard.getLogName() + " without paying its mana cost?", source, game)) {
+                game.getState().setValue("PlayFromNotOwnHandZone" + eligibleCard.getId(), Boolean.TRUE);
+                cardWasCast = controller.cast(controller.chooseAbilityForCast(eligibleCard, game, true),
+                        game, true, new ApprovingObject(source, game));
+                game.getState().setValue("PlayFromNotOwnHandZone" + eligibleCard.getId(), null);
+                if (cardWasCast) {
+                    toReveal.remove(eligibleCard);
+                }
+            }
+            controller.putCardsOnBottomOfLibrary(toReveal, game, source, false);
+        }
+        return cardWasCast;
+    }
+
+    @Override
+    public PlarggDeanOfChaosEffect copy() {
+        return new PlarggDeanOfChaosEffect(this);
+    }
+}
+
+class AugustaDeanOfOrderEffect extends OneShotEffect {
+
+    private static final FilterControlledCreaturePermanent filter = new FilterControlledCreaturePermanent("creature you control");
+
+    public AugustaDeanOfOrderEffect() {
+        super(Outcome.Benefit);
+        staticText = "tap any number of creatures you control";
+    }
+
+    public AugustaDeanOfOrderEffect(AugustaDeanOfOrderEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            TargetPermanent target = new TargetPermanent(0, 1, filter, false);
+            while (controller.canRespond()) {
+                target.clearChosen();
+                if (target.canChoose(source.getSourceId(), source.getControllerId(), game)) {
+                    Map<String, Serializable> options = new HashMap<>();
+                    options.put("UI.right.btn.text", "Tapping complete");
+                    controller.choose(outcome, target, source.getControllerId(), game, options);
+                    if (!target.getTargets().isEmpty()) {
+                        UUID creature = target.getFirstTarget();
+                        if (creature != null) {
+                            game.getPermanent(creature).tap(source, game);
+                        }
+                    } else {
+                        break;
+                    }
+                } else {
+                    break;
+                }
+            }
+            return true;
+        }
+        return false;
```

Comment:
this can be simplified by just using `TargetControlledPermanent` with an untapped filter on it to select creatures and then tapping them after selecting them

Reply:
fixed